### PR TITLE
Addresses valgrind and coverity issues

### PR DESCRIPTION
**Pull request by elliemdaw**
- Addresses memory leaks from libacvp
- Fixes static analysis errors
- Fixes build warnings

### CODE REVIEW

File: src/acvp_des.c [hunk 1]
```diff
@@ -630,6 +654,10 @@ ACVP_RESULT acvp_des_kat_handler(ACVP_CTX *ctx, JSON_Object *obj)
             key3 = (unsigned char *)json_object_get_string(testobj, "key3");
 
             /* TODO: remove this its there to work with our server */
+            if (key) {
```
Comment (bfussell) [inline src/acvp_des.c]: nit: shouldn't have to check for NULL before calling free

File: src/acvp_des.c [hunk 2]
```diff
@@ -647,7 +675,13 @@ ACVP_RESULT acvp_des_kat_handler(ACVP_CTX *ctx, JSON_Object *obj)
             if (dir == ACVP_DIR_ENCRYPT) {
                 pt = (unsigned char *)json_object_get_string(testobj, "pt");
                 iv = (unsigned char *)json_object_get_string(testobj, "iv");
-                if (!pt) return (ACVP_MALFORMED_JSON);
+                if (!pt) {
+                    if (key) {
```
Comment (bfussell) [inline src/acvp_des.c]: same here

File: src/acvp_hash.c [hunk 3]
```diff
@@ -105,6 +105,8 @@ static ACVP_RESULT acvp_hash_mct_tc(ACVP_CTX *ctx, ACVP_CAPS_LIST *cap,
     ACVP_RESULT rv;
     JSON_Value          *r_tval = NULL; /* Response testval */
     JSON_Object         *r_tobj = NULL; /* Response testobj */
+    int num_mct_outer = 100;
+    int num_mct_inner = 1000;
```
Comment (bfussell) [inline src/acvp_hash.c]: should we create a #define for these ? ACVP_HASH_MCT_INNER, OUTER ?

**Comment by elliemdaw** [comment]
@bfussell yes, spacing is yet to come. The one spot where some got cleaned up is because I couldn't figure out the memory allocation / free logic with how it was displaying in the editor, so I went ahead and cleaned that up :)

**Pull request merged**